toolkits/spotify/arcade_spotify/tools/search.py

```diff
@@ -38,8 +40,11 @@ async def search(
 
     url = get_url("search", q=q)
 
-    response = await send_spotify_request(
-        context, "GET", url, params={"q": q, "type": ",".join(types), "limit": limit}
-    )
-    response.raise_for_status()
-    return dict(response.json())
+    try:
+        response = await send_spotify_request(
+            context, "GET", url, params={"q": q, "type": ",".join(types), "limit": limit}
+        )
+        response.raise_for_status()
+        return dict(response.json())
+    except httpx.HTTPStatusError as e:
```

Comment:
The [`@tool` decorator](https://github.com/ArcadeAI/arcade-ai/blob/main/arcade/arcade/sdk/tool.py#L28-L41) already catches the error and raises a `ToolExecutionError`. The current practice for our tools is to use `raise_for_status()`, and then let the tool decorator catch the exception instead of having to explicitly try/catch inside each tool.

Reply:
This is up for debate though